tests/ops/qubit/test_observables.py

```diff
@@ -305,6 +313,61 @@ def test_hermitian_eigvals_eigvecs_same_observable_twice(
         assert len(qml.Hermitian._eigs) == 1
 
+    def test_hermitian_compute_decomposition_wire_exceptions(self):
+        """Tests user errors associated with input exceptions"""
+        single_wire_observable = np.array([[1, 1], [1, -1]]) / np.sqrt(2)
+        double_wire_observable = qml.matrix(qml.X(0) @ qml.X(1))
+
+        # test empty wire list
+        with pytest.raises(ValueError, match="At least one wire"):
+            qml.Hermitian.compute_decomposition(single_wire_observable, wires=[])
+
+        # test wrong number of wires
+        with pytest.raises(ValueError, match="Expected input matrix to have shape"):
+            qml.Hermitian.compute_decomposition(double_wire_observable, wires=[0, 1, 2])
+
+        with pytest.raises(ValueError, match="Expected input matrix to have shape"):
+            qml.Hermitian.compute_decomposition(double_wire_observable, wires="aux")
+
+        # test int as wire type
+        A_decomp = qml.Hermitian.compute_decomposition(single_wire_observable, wires=0)
+        assert np.allclose(A_decomp[0].matrix(), single_wire_observable, rtol=0)
+
+        # test str as wire type
+        A_decomp = qml.Hermitian.compute_decomposition(single_wire_observable, wires="aux")
+        assert np.allclose(A_decomp[0].matrix(), single_wire_observable, rtol=0)
+
+    def test_hermitian_compute_decomposition_inefficiency_warning(self):
+        """Tests user inefficiency warning associated with large matrix decomposition"""
+        num_wires = 8
+        observable = qml.Identity(wires=list(range(num_wires)))
+
+        with pytest.warns(
+            UserWarning, match="Decomposition may be inefficient for this large of a matrix."
+        ):
+            qml.Hermitian.compute_decomposition(
+                qml.matrix(observable), wires=list(range(num_wires))
+            )
+
+    @pytest.mark.parametrize("test_num_wires", list(range(1, 11)))
+    def test_hermitian_compute_decomposition_performance(self, test_num_wires, benchmark):
+        """Tests the performance of the compute_decomposition method of the Hermitian class.
+        Used to determine the minimum matrix dimension to raise an inefficiency warning"""
+        observable = qml.Identity(0)
+        for i in range(test_num_wires):
+            observable = observable @ qml.X(i)
+        benchmark(
+            qml.Hermitian.compute_decomposition, qml.matrix(observable), list(range(test_num_wires))
+        )
+
+    @pytest.mark.parametrize("observable", DECOMPOSITION_TEST_DATA_MULTI_WIRES)
+    def test_hermitian_compute_decomposition(self, observable):
+        """Tests that the compute_decomposition method of the Hermitian class returns the correct result."""
+        num_wires = int(np.log2(len(observable)))
+        A_decomp = qml.Hermitian.compute_decomposition(observable, wires=list(range(num_wires)))
+
+        assert np.allclose(A_decomp[0].matrix(), observable, rtol=0)
```

Comment:
Sorry if it wasn't clear in my earlier review, I meant that you can test `Hermitian.decomposition` and `Hermitian.compute_decomposition` in the same test. This is not really a blocking request though.

```suggestion
        A_decomp_static = qml.Hermitian.compute_decomposition(observable, wires=list(range(num_wires)))
        A_decomp = qml.Hermitian(observable, wires=list(range(num_wires))).decomposition()

        assert np.allclose(A_decomp_static[0].matrix(), observable, rtol=0)
        assert np.allclose(A_decomp[0].matrix(), observable, rtol=0)
```

Reply:
I see! I'll go ahead and implement that and have it ready for the team to review 🙂.